ccaas_builder/cmd/detect/main_test.go

```diff
@@ -0,0 +1,56 @@
+/*
+Copyright IBM Corp. All Rights Reserved.
+
+SPDX-License-Identifier: Apache-2.0
+*/
+
+package main
+
+import (
+	"os/exec"
+	"testing"
+	"time"
+
+	"github.com/onsi/gomega"
+	"github.com/onsi/gomega/gexec"
+)
+
+func TestArugments(t *testing.T) {
+
+	testCases := map[string]struct {
+		exitCode int
+		args     []string
+	}{
+		"toofew": {
+			exitCode: 1,
+			args:     []string{"metadatadir"},
+		},
+		"twomany": {
+			exitCode: 1,
+			args:     []string{"wibble", "wibble", "wibble"},
+		},
+		"validtype": {
+			exitCode: 0,
+			args:     []string{"na", "testdata/validtype"},
+		},
+		"wrongtype": {
+			exitCode: 1,
+			args:     []string{"na", "testdata/wrongtype"},
```

Comment:
Can you also try one with missing metadata.json, this will also show you whether the error message is good enough.